### PR TITLE
SOCKS proxy support for dialing NATS

### PR DESCRIPTION
**Pull request by philpennock**

### CODE REVIEW

File: natscontext/context.go [hunk 1]
```diff
@@ -724,3 +729,20 @@ func WithUserJWT(p string) Option {
...
 }
+
+// WithSocksProxy sets the SOCKS5 Proxy.
+// To explicitly remove an already configured proxy, use the string "none".
+func WithSocksProxy(p string) Option {
+	return func(s *settings) {
+		if p == "none" || p == "NONE" || p == "-" {
+			s.SocksProxy = ""
+		} else if p != "" {
+			s.SocksProxy = p
```
Comment (ripienaar) [inline natscontext/context.go:740]: maybe url parse here like you would do in the proxy setup also. This will give early user feedback

Comment (philpennock) [reply natscontext/context.go:740]: I like the idea but I can't see a library-sensible way to return a diagnostic here.  I don't think we should emit to stderr in a library, and there's no way to return an error, or stash a warning for later printing.

So for now I've taken the LGTM and merged as-is, but if you can suggest a clean way to add extra diagnostics here, I'll see about implementing it.

Comment (ripienaar) [reply natscontext/context.go:740]: yeah fair I didnt see the option type doesnt have error here, I usually add that so missed it :) thank you

Comment (ripienaar) [reply natscontext/context.go:740]: today this requires like `socks5://blah:port`, we might assume socks and default port when someone gives just `blah` or `blah:port`?

**Pull request merged**